lua/telescope/actions/init.lua

```diff
@@ -805,6 +805,213 @@ actions.cycle_previewers_prev = function(prompt_bufnr)
+      if string.find(vim.api.nvim_buf_get_name(buf), "_TelescopePickersKeymap") then
+        return buf
+      end
+    end
+  end
+
+  -- close on repeated keypress
+  local km_buf = find_keymap_buf()
+  if km_buf ~= nil then
+    vim.api.nvim_buf_delete(km_buf, { force = true, unload = false })
+    local win_ids = vim.fn.win_findbuf(km_buf)
+    for _, win_id in ipairs(win_ids) do
+      pcall(vim.api.nvim_win_close, win_id, true)
+    end
+    return
+  end
+
+  local entry_display = require "telescope.pickers.entry_display"
+  local displayer = entry_display.create {
+    separator = utils.get_default(opts.seperator, " -> "),
+    items = {
+      { width = opts.mode_with },
+      { width = opts.keybind_width },
+      { width = opts.name_width },
+    },
+  }
+
+  local make_display = function(mapping)
+    return displayer {
+      { mapping.mode, utils.get_default(opts.mode_hl, "TelescopeResultsConstant") },
+      { mapping.keybind, utils.get_default(opts.keybind_hl, "TelescopeResultsVariable") },
+      { mapping.name, utils.get_default(opts.name_hl, "TelescopeResultsFunction") },
+    }
+  end
+
+  local prompt_buf_mappings = __TelescopeKeymapStore[prompt_bufnr]
+
+  local mappings = {}
+  for _, v in pairs(prompt_buf_mappings) do
+    if type(v) == "table" then
+      local name = ""
+      for _, action in ipairs(v) do
+        if type(action) == "string" then
+          name = name == "" and action or name .. " + " .. action
+        end
+      end
+      table.insert(mappings, { mode = v.mode, keybind = v.keybind, name = name })
+    end
+  end
+  if opts.only_show_current_mode then
+    local mode = vim.api.nvim_get_mode().mode
+    mappings = vim.tbl_filter(function(t)
+      return t.mode == mode
+    end, mappings)
+  end
+
+  table.sort(mappings, function(x, y)
+    if x.name < y.name then
+      return true
+    elseif x.name == y.name then
+      -- show normal mode as the standard mode first
+      if x.mode > y.mode then
+        return true
+      else
+        -- loosely prefers simpler mappings
+        if x.keybind < y.keybind then
+          return true
+        end
+        return false
+      end
+    else
+      return false
+    end
+  end)
+
+  local entry_width = #opts.column_padding
+    + opts.mode_width
+    + opts.keybind_width
+    + opts.name_width
+    + (3 * #opts.separator)
+  local num_total_columns = math.floor((vim.o.columns - #opts.column_indent) / entry_width)
+  opts.num_rows = math.ceil(math.min(#mappings / num_total_columns, opts.max_height * vim.o.lines))
+  local total_available_entries = opts.num_rows * num_total_columns
+
+  km_buf = vim.api.nvim_create_buf(false, false)
+  vim.api.nvim_buf_set_name(km_buf, "_TelescopePickersKeymap")
+
+  --   place top or bottom
+  local prompt_win = action_state.get_current_picker(prompt_bufnr).prompt_win
+  local prompt_row = vim.api.nvim_win_get_position(prompt_win)[1]
+  local prompt_pos = prompt_row <= 0.4 * vim.o.lines
```

Comment:
I think it is best to use `0.5` instead of `0.4` here.

Reply:
Ok, no strong feelings here.

Reply:
My thinking is that if this is a check for the prompt being in the top half of the screen, it should check the top *half*. Probably wouldn't make a difference most of the time, but makes the functionality easier to understand IMO.

Reply:
My apologies, I'm too tired, 1am my current time zone, I shouldn't be replying 😅 I initially skimmed to quickly and thought you were referring to the default value of `opts.max_height`, which I would not have had strong feelings to change.

Here, it's just a typo, you are of course right. Thanks for reiterating :)

Reply:
No worries 🙂